File: mobile/lib/domain/models/channel.dart
class Channel {
  final String id;
  final String name;
  final String logoUrl;
  final String categoryId;
  final String languageCode;
  final String countryCode;
  final String providerType;

  const Channel({
    required this.id,
    required this.name,
    required this.logoUrl,
    required this.categoryId,
    required this.languageCode,
    required this.countryCode,
    required this.providerType,
  });

  factory Channel.fromJson(Map<String, dynamic> json) {
    return Channel(
      id: json['ID'] as String? ?? '',
      name: json['Name'] as String? ?? 'Unknown',
      logoUrl: json['LogoURL'] as String? ?? '',
      categoryId: json['CategoryID'] as String? ?? '',
      languageCode: json['LanguageCode'] as String? ?? '',
      countryCode: json['CountryCode'] as String? ?? '',
      providerType: json['ProviderType'] as String? ?? '',
    );
  }
}

File: gateway/internal/domain/channel.go
package domain

import (
	"fmt"
	"time"
)

type ChannelID string
type ProviderType string

const (
	ProviderOpenSource ProviderType = "opensource"
)

type Channel struct {
	ID ChannelID
	// TvgID es el identificador XMLTV (tvg-id del M3U); une el canal con su EPG.
	TvgID        string
	Name         string
	LogoURL      string
	CategoryID   string
	LanguageCode string // ISO 639-1
	CountryCode  string // ISO 3166-1 alpha-2
	ProviderID   string
	// Salud agregada de los streams del canal, rellenada por FindFiltered
	// para que la lista pinte el indicador sin N+1 a /channels/{id}/health.
	// Alive nil = ningún stream chequeado aún.
	Alive        *bool
	LatencyMs    int64 // mejor latencia entre streams vivos; 0 si no aplica
	ProviderType ProviderType
	IsAdult      bool
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

func (c Channel) Validate() error {
	if c.ID == "" {
		return fmt.Errorf("channel.Validate: ID requerido")
	}
	if c.Name == "" {
		return fmt.Errorf("channel.Validate: Name requerido")
	}
	if c.ProviderType == "" {
		return fmt.Errorf("channel.Validate: ProviderType requerido")
	}
	return nil
}
